packages/dashboard/src/components/dispensers-panel.tsx

```diff
@@ -18,6 +19,27 @@ export const DispenserPanel = React.memo((props: DispenserPanelProps) => {
   const dispenserRefs = React.useRef<Record<string, HTMLElement | null>>({});
   const [expanded, setExpanded] = React.useState<Record<string, boolean>>({});
 
+  const dispenserResourcesContext = React.useContext(ResourcesContext)?.dispensers;
+  const dispenserList = dispenserResourcesContext
+    ? Object.keys(dispenserResourcesContext.dispensers)
+    : [];
+  const updateDispenserStates: Record<string, RomiCore.DispenserState> = {};
+
+  dispenserList.forEach((dispenser) => {
+    if (!dispenserStates.hasOwnProperty(dispenser)) {
+      const unknownStates: RomiCore.DispenserState = {
+        guid: dispenser,
+        mode: 2,
+        request_guid_queue: [],
+        seconds_remaining: 0,
+        time: { sec: 0, nanosec: 0 },
+      };
+      updateDispenserStates[dispenser] = unknownStates;
+    } else {
+      updateDispenserStates[dispenser] = dispenserStates[dispenser];
+    }
+  });
```

Comment:
we can put this inside a useMemo too. So it'll change only if `dispenserList` changes.

we can change this name `updateDispenserStates` to a storage kind of name and not as an action. Maybe, `newDispenserStates` or `updatedDispenseStates` or `dispenserStatesUpdated`

Reply:
Instead of hardcoding something as an unknown state, we should have the UI display something different indicating the state is unknown.